app-arch/zstd/patches/zstd-1.3.5.patchset

```diff
@@ -0,0 +1,126 @@
+ # make install is validated only for Linux, macOS, BSD, Hurd and Solaris targets
+ #-----------------------------------------------------------------------------
+-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
++ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS Haiku))
+ 
+ DESTDIR     ?=
+ # directory variables : GNU conventions prefer lowercase
+diff --git a/programs/Makefile b/programs/Makefile
+index 4202764..579a349 100644
+--- a/programs/Makefile
++++ b/programs/Makefile
+@@ -245,7 +245,7 @@ preview-man: clean-man man
+ #-----------------------------------------------------------------------------
+ # make install is validated only for Linux, macOS, BSD, Hurd and Solaris targets
+ #-----------------------------------------------------------------------------
+-ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS))
++ifneq (,$(filter $(shell uname),Linux Darwin GNU/kFreeBSD GNU OpenBSD FreeBSD NetBSD DragonFly SunOS Haiku))
+ 
+ .PHONY: list
+ list:
+diff --git a/programs/platform.h b/programs/platform.h
+index a550eb1..55aa4e3 100644
+--- a/programs/platform.h
++++ b/programs/platform.h
+@@ -70,9 +70,10 @@ extern "C" {
+ *  PLATFORM_POSIX_VERSION >= 1 is equal to found _POSIX_VERSION
+ ***************************************************************/
+ #if !defined(_WIN32) && (defined(__unix__) || defined(__unix) || (defined(__APPLE__) && defined(__MACH__)) /* UNIX-like OS */ \
+-   || defined(__midipix__) || defined(__VMS))
++   || defined(__midipix__) || defined(__VMS) || defined(__HAIKU__))
+ #  if (defined(__APPLE__) && defined(__MACH__)) || defined(__SVR4) || defined(_AIX) || defined(__hpux) /* POSIX.1-2001 (SUSv3) conformant */ \
+-     || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)  /* BSD distros */
++     || defined(__DragonFly__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)  /* BSD distros */ \
++     || defined(__HAIKU__)
+ #    define PLATFORM_POSIX_VERSION 200112L
+ #  else
+ #    if defined(__linux__) || defined(__linux)
+diff --git a/programs/util.h b/programs/util.h
+index 4392a5b..26299cf 100644
+--- a/programs/util.h
++++ b/programs/util.h
+@@ -64,7 +64,7 @@ extern "C" {
+ #elif PLATFORM_POSIX_VERSION >= 0 /* Unix-like operating system */
+ #  include <unistd.h>
+ #  include <sys/resource.h> /* setpriority */
+-#  if defined(PRIO_PROCESS)
++#  if defined(PRIO_PROCESS) && !defined(__HAIKU__)
+ #    define SET_REALTIME_PRIORITY setpriority(PRIO_PROCESS, 0, -20)
+ #  else
+ #    define SET_REALTIME_PRIORITY /* disabled */
+-- 
+2.19.0
+
+
+From a2e6a422668cad4591e5677e4cb275bf966503c8 Mon Sep 17 00:00:00 2001
+From: fbrosson <fbrosson@localhost>
+Date: Wed, 19 Sep 2018 11:38:30 +0200
+Subject: gcc2: Skip PREFETCH() to avoid undefined ref to __builtin_prefetch
```

Comment:
This has been upstreamed in facebook/zstd/pull/1139 and got merged before the release of `1.3.5`.

BTW, I did not test 1.3.5 yet but I'm quite sure it will be fine to drop `1.3.4` and turn this PR into a real _"version bump to 1.3.5"_ instead of a _"add recipe for version 1.3.5"_.

Reply:
Need to recheck, but iirc it wasn't included in the source ....

Reply:
Well, what got merged was a better fix than the one we had in our patchset.
The reason why the old patch still applies nicely is because the patch which got merged fixes the issue in a different file.
(So I'm confident that when you will recheck you will find that it is OK to drop this patch chunk :)

Reply:
Just checked, it isn't in 1.3.5 archive ...

Reply:
Missed the last comment before I checked, will check with dropping that chunk ;)